webapp/packages/plugin-authentication-administration/src/Administration/Users/UserForm/AdministrationUserFormDeleteButton.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { UsersResource } from '@cloudbeaver/core-authentication';
+import { Button, ButtonProps, useTranslate } from '@cloudbeaver/core-blocks';
+import { useService } from '@cloudbeaver/core-di';
+import { CommonDialogService, DialogueStateResult } from '@cloudbeaver/core-dialogs';
+
+import { AdministrationUsersManagementService } from '../../../AdministrationUsersManagementService';
+import { DeleteUserDialog } from './DeleteUserDialog';
+import { DisableUserDialog } from './DisableUserDialog';
+
+interface Props extends ButtonProps {
+  userId: string;
+  enabled: boolean;
+}
+
+export const AdministrationUserFormDeleteButton: React.FC<Props> = function AdministrationUserFormDeleteButton({ userId, enabled, ...rest }) {
+  const translate = useTranslate();
+  const commonDialogService = useService(CommonDialogService);
+  const administrationUsersManagementService = useService(AdministrationUsersManagementService);
+  const usersResource = useService(UsersResource);
+
+  const userManagementDisabled = administrationUsersManagementService.externalUserProviderEnabled;
+  const deleteDisabled = usersResource.isActiveUser(userId) || userManagementDisabled;
+
+  if (deleteDisabled) {
+    return null;
+  }
+
+  async function deleteUser() {
+    if (enabled) {
+      const result = await commonDialogService.open(DisableUserDialog, {
+        userId,
+      });
+
+      if (result === DialogueStateResult.Rejected) {
+        return;
+      }
+    }
+
+    await commonDialogService.open(DeleteUserDialog, {
+      userId,
+    });
```

Comment:
please use callback to display delete user dialog, because current behaviour is hard to understand

dialog itself can be rejected or resolved or return ressult. if we display dialog like "DeleteConfirmation" than we can cleraly understad result because it can be `DialogueStateResult.Rejected` (closed/cancel) or `DialogueStateResult.Resolved` (agreed / deleted)

In this function is hard to understand the workflow: 
Why we want to open "DeleteUserDialod" if `DisableUserDialog` was resolved (is user disabled?)